### PR TITLE
Implement Transer api call to Air Call

### PR DESCRIPTION
**Pull request by hockeybuggy**
Previously this function would do nothing. Now this function will make
a call to AirCall. In order to make this call it needs two environment
variables `AIRCALL_ID` and `AIRCALL_TOKEN`. These values have been set
in prod and staging.

This PR also setups up the `app.test.js` as end to end tests. These only mock the call to Axios, but everything else is "real".

In order to make this PR landable I have set the bypass on both staging and prod. I am not really sure what to do about staging for this service since there is no "airtable staging" and any webhooks sent to that service would do the same thing that prod would do.

### CODE REVIEW

File: src/app.test.js [hunk 1]
```diff
@@ -32,32 +37,66 @@ describe('aircall webhook', () => {
...
-  });
-
-  test('inbound call.created', async () => {
-    const result = await request(app).post(
-      '/aircall/calls',
-      inboundCallCreated
+    process.env.AIRCALL_ID = 'some_id';
+    process.env.AIRCALL_TOKEN = 'some_token';
+    axios.post.mockClear();
+    axios.post.mockImplementation(() =>
```
Comment (gavinsharp) [inline src/app.test.js:49]: (I vaguely recall making this same suggestion before and you disagreeing, but I will try again anyways 😄)
I find the equivalent [`mockResolvedValue`](https://jestjs.io/docs/en/mock-function-api#mockfnmockresolvedvaluevalue) helper slightly easier to understand when reading test logic.

File: src/app.test.js [hunk 2]
```diff
@@ -32,32 +37,66 @@ describe('aircall webhook', () => {
...
+      Promise.resolve({ data: { ok: ':thumbsup:' } })
     );
-    // TODO assert side effects happen
-    expect(result.statusCode).toEqual(200);
   });
 
-  test('inbound call.created - With bypass', async () => {
-    process.env.PROPERLY_BYPASS = 'true'; // Note: any non `false` value should work.
+  describe('inbound call.created', () => {
+    test('within bussiness hours', async () => {
```
Comment (gavinsharp) [inline src/app.test.js:55]:
```suggestion
    test('within business hours', async () => {
```

File: src/services.test.js [hunk 3]
```diff
@@ -19,3 +24,67 @@ describe('isOutsideOfBusinessHours', () => {
...
+      started_at: 1584998199,
+      answered_at: null,
+      ended_at: null,
+      duration: 0,
+      number: {
+        id: PHONE_NUMBER_AIRCALL_IDS.MAIN,
+      },
+    };
+
+    expect(axios.post).not.toHaveBeenCalled();
```
Comment (leemadeleine) [inline src/services.test.js:74]: What is this assertion testing? It isn't clear to me how `axois.post` would be called before calling `transferCall`

File: src/app.test.js (hunk 1 above)
Comment (hockeybuggy) [reply src/app.test.js:49]: I think you might be referring to this PR thread: https://github.com/GoProperly/www/pull/208#discussion_r459639000 If that is the thread: I would characterize that is asking why you preferred your suggestion over alternatives.

`mockResolvedValue` seems like it makes things more concise, but may require more "jest knowledge" to know what's happening (such as how does it differ from `mockImplementation` or `mockReturn`). The docs are good, and once someone knows what it is I expect it to be more clear. Updating.

File: src/services.test.js (hunk 3 above)
Comment (hockeybuggy) [reply src/services.test.js:74]: This is a "precondition assertion". It's intention is to confirm that "the world" is in a particular state before getting into a part of the test that is exercising the "subject under test".

Personal preference: There is a particular structure I prefer when writing a test. My preference is for tests to have a structure similar to this (which each section being a section of code with no blank lines):

```
test(() => {
  // Setup

  // [optional] Precondition assertions

  // Call the subject under test

  // Post condition assertions

  // [optional] Cleanup
});
```

File: src/app.test.js (hunk 1 above)
Comment (hockeybuggy) [reply src/app.test.js]: Changed in 4f2e5ba

File: src/services.test.js (hunk 3 above)
Comment (leemadeleine) [reply src/services.test.js:74]: Thanks! That was a great explanation

**Pull request merged**